In [4]:
import pandas as pd

df = pd.read_csv('train_transaction.csv', nrows=10000)
print(df.shape)
print(df.columns.tolist())

(10000, 394)
['TransactionID', 'isFraud', 'TransactionDT', 'TransactionAmt', 'ProductCD', 'card1', 'card2', 'card3', 'card4', 'card5', 'card6', 'addr1', 'addr2', 'dist1', 'dist2', 'P_emaildomain', 'R_emaildomain', 'C1', 'C2', 'C3', 'C4', 'C5', 'C6', 'C7', 'C8', 'C9', 'C10', 'C11', 'C12', 'C13', 'C14', 'D1', 'D2', 'D3', 'D4', 'D5', 'D6', 'D7', 'D8', 'D9', 'D10', 'D11', 'D12', 'D13', 'D14', 'D15', 'M1', 'M2', 'M3', 'M4', 'M5', 'M6', 'M7', 'M8', 'M9', 'V1', 'V2', 'V3', 'V4', 'V5', 'V6', 'V7', 'V8', 'V9', 'V10', 'V11', 'V12', 'V13', 'V14', 'V15', 'V16', 'V17', 'V18', 'V19', 'V20', 'V21', 'V22', 'V23', 'V24', 'V25', 'V26', 'V27', 'V28', 'V29', 'V30', 'V31', 'V32', 'V33', 'V34', 'V35', 'V36', 'V37', 'V38', 'V39', 'V40', 'V41', 'V42', 'V43', 'V44', 'V45', 'V46', 'V47', 'V48', 'V49', 'V50', 'V51', 'V52', 'V53', 'V54', 'V55', 'V56', 'V57', 'V58', 'V59', 'V60', 'V61', 'V62', 'V63', 'V64', 'V65', 'V66', 'V67', 'V68', 'V69', 'V70', 'V71', 'V72', 'V73', 'V74', 'V75', 'V76', 'V77', 'V78', 'V79', 'V8

In [5]:
!pip install tensorflow optuna mlflow lime -q

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 49.7/49.7 kB 3.6 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 275.7/275.7 kB 14.1 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 50.5/50.5 kB 3.8 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 43.7/43.7 kB 3.6 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 425.6/425.6 kB 19.8 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.6/12.6 MB 83.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.5/3.5 MB 88.8 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.7/1.7 MB 63.8 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.7/4.7 MB 60.8 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 147.8/147.8 kB 13.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 114.9/114.9 kB 9.8 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 212.0/212.0 kB 16.9 MB/s eta 0:00:00
   ━━

In [6]:
import tensorflow as tf
import optuna
import mlflow
import lime

print("TensorFlow :", tf.__version__)
print("Optuna     :", optuna.__version__)
print("MLflow     :", mlflow.__version__)
print("LIME       :", lime.__version__ if hasattr(lime, "__version__") else "Installed")

TensorFlow : 2.20.0
Optuna     : 4.9.0
MLflow     : 3.14.0
LIME       : Installed


In [7]:
import os

print(os.listdir())

['.config', 'train_transaction.csv', 'sample_data']


In [8]:
import pandas as pd

df = pd.read_csv("train_transaction.csv")

print(df.shape)


(590540, 394)


In [9]:
df["isFraud"].value_counts()

,count
isFraud,
0,569877
1,20663


In [10]:
df.isnull().sum().sort_values(ascending=False).head(20)

,0
dist2,552913
D7,551623
D13,528588
D14,528353
D12,525823
D6,517353
D9,515614
D8,515614
V153,508595
V149,508595


In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 590540 entries, 0 to 590539
Columns: 394 entries, TransactionID to V339
dtypes: float64(376), int64(4), object(14)
memory usage: 1.7+ GB


In [1]:
import tensorflow as tf
print(tf.config.list_physical_devices('GPU'))

[PhysicalDevice(name='/physical_device:GPU:0', device_type='GPU')]


In [4]:
import os
print(os.listdir())

['.config', 'sample_data']


In [6]:
from google.colab import files

uploaded = files.upload()

Saving train_transaction.csv to train_transaction.csv


In [7]:
import pandas as pd

df = pd.read_csv("train_transaction.csv")

print(df.shape)


(590540, 394)


In [8]:
df["isFraud"].value_counts()

,count
isFraud,
0,569877
1,20663


In [9]:
print(df["isFraud"].value_counts(normalize=True) * 100)

isFraud
0    96.500999
1     3.499001
Name: proportion, dtype: float64


In [10]:
df.isnull().sum().sort_values(ascending=False).head(20)

,0
dist2,552913
D7,551623
D13,528588
D14,528353
D12,525823
D6,517353
D8,515614
D9,515614
V162,508595
V142,508595


In [11]:
X = df.drop(["isFraud", "TransactionID"], axis=1)
y = df["isFraud"]

In [12]:
num_cols = X.select_dtypes(include=['int64','float64']).columns
cat_cols = X.select_dtypes(include=['object']).columns

print("Numerical:", len(num_cols))
print("Categorical:", len(cat_cols))

Numerical: 378
Categorical: 14


In [13]:
from sklearn.compose import ColumnTransformer
from sklearn.pipeline import Pipeline

from sklearn.impute import SimpleImputer
from sklearn.preprocessing import StandardScaler
from sklearn.preprocessing import OrdinalEncoder

In [14]:
num_transformer = Pipeline([
    ('imputer', SimpleImputer(strategy='median')),
    ('scaler', StandardScaler())
])

cat_transformer = Pipeline([
    ('imputer', SimpleImputer(strategy='most_frequent')),
    ('encoder',
     OrdinalEncoder(
         handle_unknown='use_encoded_value',
         unknown_value=-1
     ))
])

In [15]:
preprocessor = ColumnTransformer([
    ('num', num_transformer, num_cols),
    ('cat', cat_transformer, cat_cols)
])

In [16]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.2,
    random_state=42,
    stratify=y
)

In [1]:
import pandas as pd

df = pd.read_csv(
    "train_transaction.csv",
    nrows=100000
)

print(df.shape)

(100000, 394)


In [2]:
X = df.drop(["isFraud","TransactionID"], axis=1)
y = df["isFraud"]

In [3]:
missing_ratio = X.isnull().mean()

drop_cols = missing_ratio[missing_ratio > 0.8].index

X = X.drop(columns=drop_cols)

print("Dropped:", len(drop_cols))
print(X.shape)

Dropped: 9
(100000, 383)


In [4]:
df = pd.read_csv(
    "train_transaction.csv",
    nrows=50000
)

In [5]:
import os
print(os.listdir())

['.config', 'train_transaction.csv', 'sample_data']


In [6]:
X = df.drop(["isFraud","TransactionID"], axis=1)
y = df["isFraud"]

print(X.shape)
print(y.shape)

(50000, 392)
(50000,)


In [7]:
num_cols = X.select_dtypes(include=['int64','float64']).columns
cat_cols = X.select_dtypes(include=['object']).columns

print("Numerical:", len(num_cols))
print("Categorical:", len(cat_cols))

Numerical: 378
Categorical: 14


In [8]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.2,
    random_state=42,
    stratify=y
)

print(X_train.shape)
print(X_test.shape)

(40000, 392)
(10000, 392)


In [9]:
X_train[num_cols] = X_train[num_cols].fillna(
    X_train[num_cols].median()
)

X_test[num_cols] = X_test[num_cols].fillna(
    X_train[num_cols].median()
)

In [10]:
for col in cat_cols:
    X_train[col] = X_train[col].fillna("missing")
    X_test[col] = X_test[col].fillna("missing")

In [11]:
from sklearn.preprocessing import OrdinalEncoder

encoder = OrdinalEncoder(
    handle_unknown='use_encoded_value',
    unknown_value=-1
)

X_train[cat_cols] = encoder.fit_transform(
    X_train[cat_cols]
)

X_test[cat_cols] = encoder.transform(
    X_test[cat_cols]
)

In [12]:
from sklearn.preprocessing import StandardScaler

scaler = StandardScaler()

X_train = scaler.fit_transform(X_train)
X_test = scaler.transform(X_test)

print(X_train.shape)
print(X_test.shape)

(40000, 392)
(10000, 392)


In [13]:
import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Dropout

In [14]:
model = Sequential([
    Dense(256, activation='relu',
          input_shape=(X_train.shape[1],)),
    Dropout(0.3),

    Dense(128, activation='relu'),
    Dropout(0.3),

    Dense(64, activation='relu'),

    Dense(1, activation='sigmoid')
])

/usr/local/lib/python3.12/dist-packages/keras/src/layers/core/dense.py:106: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


In [15]:
model.compile(
    optimizer='adam',
    loss='binary_crossentropy',
    metrics=[
        tf.keras.metrics.AUC(name='auc'),
        'accuracy'
    ]
)

In [16]:
history = model.fit(
    X_train,
    y_train,
    validation_split=0.2,
    epochs=10,
    batch_size=512
)

Epoch 1/10
63/63 ━━━━━━━━━━━━━━━━━━━━ 13s 68ms/step - accuracy: 0.9532 - auc: 0.6697 - loss: 0.1644 - val_accuracy: 0.9776 - val_auc: 0.8318 - val_loss: 0.0886
Epoch 2/10
63/63 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.9748 - auc: 0.8173 - loss: 0.1004 - val_accuracy: 0.9785 - val_auc: 0.8624 - val_loss: 0.0823
Epoch 3/10
63/63 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.9758 - auc: 0.8544 - loss: 0.0910 - val_accuracy: 0.9800 - val_auc: 0.8706 - val_loss: 0.0776
Epoch 4/10
63/63 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.9766 - auc: 0.8648 - loss: 0.0885 - val_accuracy: 0.9801 - val_auc: 0.8709 - val_loss: 0.0786
Epoch 5/10
63/63 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.9777 - auc: 0.8685 - loss: 0.0855 - val_accuracy: 0.9800 - val_auc: 0.8830 - val_loss: 0.0786
Epoch 6/10
63/63 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.9782 - auc: 0.8791 - loss: 0.0827 - val_accuracy: 0.9801 - val_auc: 0.8809 - val_loss: 0.0767
Epoch 7/10
63/63 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step 

In [17]:
y_prob = model.predict(X_test)

y_pred = (y_prob > 0.5).astype(int)

313/313 ━━━━━━━━━━━━━━━━━━━━ 3s 8ms/step


In [18]:
from sklearn.metrics import accuracy_score
from sklearn.metrics import precision_score
from sklearn.metrics import recall_score
from sklearn.metrics import f1_score
from sklearn.metrics import roc_auc_score
from sklearn.metrics import confusion_matrix

print("Accuracy :", accuracy_score(y_test, y_pred))
print("Precision:", precision_score(y_test, y_pred))
print("Recall   :", recall_score(y_test, y_pred))
print("F1 Score :", f1_score(y_test, y_pred))
print("ROC AUC  :", roc_auc_score(y_test, y_prob))

print(confusion_matrix(y_test, y_pred))

Accuracy : 0.9807
Precision: 0.8421052631578947
Recall   : 0.35424354243542433
F1 Score : 0.4987012987012987
ROC AUC  : 0.889780960714325
[[9711   18]
 [ 175   96]]


## Analisis Hasil Fraud Detection

Berdasarkan hasil pengujian, model Deep Learning memperoleh nilai Accuracy sebesar 98,07%, Precision sebesar 84,21%, Recall sebesar 35,42%, F1-Score sebesar 49,87%, dan ROC-AUC sebesar 88,98%.

Nilai Accuracy yang tinggi menunjukkan bahwa model mampu mengklasifikasikan sebagian besar transaksi dengan benar. Namun, karena dataset fraud detection memiliki distribusi kelas yang tidak seimbang, nilai Accuracy tidak dapat dijadikan satu-satunya indikator keberhasilan model.

Nilai Precision sebesar 84,21% menunjukkan bahwa sebagian besar transaksi yang diprediksi sebagai fraud memang benar merupakan transaksi fraud. Akan tetapi, nilai Recall sebesar 35,42% menunjukkan bahwa masih terdapat sejumlah transaksi fraud yang belum berhasil terdeteksi oleh model.

Berdasarkan confusion matrix, model berhasil mengidentifikasi 96 transaksi fraud dengan benar dan 9.711 transaksi normal dengan benar. Namun, masih terdapat 175 transaksi fraud yang salah diklasifikasikan sebagai transaksi normal. Kondisi ini menunjukkan bahwa model masih cenderung lebih baik dalam mengenali transaksi normal dibandingkan transaksi fraud.

Secara keseluruhan, model telah menunjukkan performa yang cukup baik dengan nilai ROC-AUC sebesar 88,98%, yang menandakan kemampuan model dalam membedakan transaksi fraud dan non-fraud sudah baik. Meskipun demikian, peningkatan pada nilai Recall masih diperlukan agar lebih banyak transaksi fraud dapat terdeteksi.


## Kesimpulan

Model Deep Learning berhasil diterapkan untuk mendeteksi transaksi fraud menggunakan dataset train_transaction.csv. Setelah dilakukan preprocessing data, penanganan missing value, encoding fitur kategorikal, dan normalisasi data, model memperoleh Accuracy sebesar 98,07%, Precision sebesar 84,21%, Recall sebesar 35,42%, F1-Score sebesar 49,87%, dan ROC-AUC sebesar 88,98%.

Hasil tersebut menunjukkan bahwa model mampu membedakan transaksi fraud dan non-fraud dengan cukup baik. Namun, kemampuan model dalam mendeteksi seluruh transaksi fraud masih perlu ditingkatkan karena nilai Recall masih relatif rendah. Pengembangan lebih lanjut dapat dilakukan dengan hyperparameter tuning menggunakan Optuna dan penanganan data tidak seimbang yang lebih baik.


In [1]:
!pip install optuna

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 425.6/425.6 kB 15.1 MB/s eta 0:00:00


In [2]:
import optuna
import tensorflow as tf

In [3]:
def objective(trial):

    units1 = trial.suggest_int("units1", 64, 256)
    units2 = trial.suggest_int("units2", 32, 128)

    dropout = trial.suggest_float(
        "dropout",
        0.1,
        0.5
    )

    lr = trial.suggest_float(
        "learning_rate",
        1e-4,
        1e-2,
        log=True
    )

    model = tf.keras.Sequential([
        tf.keras.layers.Dense(
            units1,
            activation="relu",
            input_shape=(X_train.shape[1],)
        ),

        tf.keras.layers.Dropout(dropout),

        tf.keras.layers.Dense(
            units2,
            activation="relu"
        ),

        tf.keras.layers.Dense(
            1,
            activation="sigmoid"
        )
    ])

    model.compile(
        optimizer=tf.keras.optimizers.Adam(
            learning_rate=lr
        ),
        loss="binary_crossentropy",
        metrics=["accuracy"]
    )

    history = model.fit(
        X_train,
        y_train,
        validation_split=0.2,
        epochs=3,
        batch_size=512,
        verbose=0
    )

    return max(
        history.history["val_accuracy"]
    )

In [5]:
print('X_train' in globals())
print('y_train' in globals())

False
False


In [14]:
import pandas as pd

print("pandas siap")

pandas siap


In [16]:
import os
print(os.listdir())

['.config', 'sample_data']


In [18]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [19]:
print(os.listdir('/content/drive/MyDrive/UAS'))

['midterm-regresi-dataset.csv', 'train_transaction.csv']


In [20]:
import pandas as pd

df = pd.read_csv(
    '/content/drive/MyDrive/UAS/train_transaction.csv',
    nrows=50000
)

print(df.shape)

(50000, 394)


In [21]:
X = df.drop(
    ["isFraud", "TransactionID"],
    axis=1
)

y = df["isFraud"]

print(X.shape)
print(y.shape)

(50000, 392)
(50000,)


In [22]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.2,
    random_state=42,
    stratify=y
)

print(X_train.shape)
print(X_test.shape)

(40000, 392)
(10000, 392)


In [24]:
num_cols = X_train.select_dtypes(
    include=['int64','float64']
).columns

cat_cols = X_train.select_dtypes(
    include=['object']
).columns

print("Numerical:", len(num_cols))
print("Categorical:", len(cat_cols))

Numerical: 378
Categorical: 14


In [25]:
from sklearn.impute import SimpleImputer

num_imputer = SimpleImputer(strategy='median')

X_train[num_cols] = num_imputer.fit_transform(
    X_train[num_cols]
)

X_test[num_cols] = num_imputer.transform(
    X_test[num_cols]
)

In [26]:
from sklearn.preprocessing import OrdinalEncoder

encoder = OrdinalEncoder(
    handle_unknown='use_encoded_value',
    unknown_value=-1
)

X_train[cat_cols] = encoder.fit_transform(
    X_train[cat_cols]
)

X_test[cat_cols] = encoder.transform(
    X_test[cat_cols]
)

In [27]:
from sklearn.preprocessing import StandardScaler

scaler = StandardScaler()

X_train = scaler.fit_transform(X_train)
X_test = scaler.transform(X_test)

print(X_train.shape)
print(X_test.shape)

(40000, 392)
(10000, 392)


In [28]:
!pip install optuna -q

In [29]:
import optuna
import tensorflow as tf

In [30]:
def objective(trial):

    units1 = trial.suggest_int("units1", 64, 256)
    units2 = trial.suggest_int("units2", 32, 128)
    dropout = trial.suggest_float("dropout", 0.1, 0.5)

    model = tf.keras.Sequential([
        tf.keras.layers.Dense(
            units1,
            activation="relu",
            input_shape=(X_train.shape[1],)
        ),

        tf.keras.layers.Dropout(dropout),

        tf.keras.layers.Dense(
            units2,
            activation="relu"
        ),

        tf.keras.layers.Dense(
            1,
            activation="sigmoid"
        )
    ])

    model.compile(
        optimizer="adam",
        loss="binary_crossentropy",
        metrics=["accuracy"]
    )

    history = model.fit(
        X_train,
        y_train,
        validation_split=0.2,
        epochs=3,
        batch_size=512,
        verbose=0
    )

    return max(history.history["val_accuracy"])

In [ ]:
study = optuna.create_study(
    direction="maximize"
)

study.optimize(
    objective,
    n_trials=10
)

[I 2026-06-23 23:03:44,267] A new study created in memory with name: no-name-5cdbf778-6bb9-4906-9a78-6d1c7c7e9166
/usr/local/lib/python3.12/dist-packages/keras/src/layers/core/dense.py:106: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)
[I 2026-06-23 23:03:54,878] Trial 0 finished with value: 0.9743750095367432 and parameters: {'units1': 108, 'units2': 51, 'dropout': 0.11505569200175247}. Best is trial 0 with value: 0.9743750095367432.
[I 2026-06-23 23:04:01,776] Trial 1 finished with value: 0.9743750095367432 and parameters: {'units1': 176, 'units2': 49, 'dropout': 0.42666054199110803}. Best is trial 0 with value: 0.9743750095367432.
[I 2026-06-23 23:04:06,896] Trial 2 finished with value: 0.9743750095367432 and parameters: {'units1': 141, 'units2': 116, 'dropout': 0.316964444809

In [32]:
print("Best Parameters:")
print(study.best_params)

print("\nBest Accuracy:")
print(study.best_value)

Best Parameters:
{'units1': 108, 'units2': 51, 'dropout': 0.11505569200175247}

Best Accuracy:
0.9743750095367432


In [33]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.2,
    random_state=42,
    stratify=y
)

print(X_train.shape)
print(X_test.shape)

(40000, 392)
(10000, 392)


In [34]:
X_train[num_cols] = X_train[num_cols].fillna(
    X_train[num_cols].median()
)

X_test[num_cols] = X_test[num_cols].fillna(
    X_train[num_cols].median()
)

In [35]:
for col in cat_cols:
    X_train[col] = X_train[col].fillna("missing")
    X_test[col] = X_test[col].fillna("missing")

In [36]:
from sklearn.preprocessing import OrdinalEncoder

encoder = OrdinalEncoder(
    handle_unknown='use_encoded_value',
    unknown_value=-1
)

X_train[cat_cols] = encoder.fit_transform(
    X_train[cat_cols]
)

X_test[cat_cols] = encoder.transform(
    X_test[cat_cols]
)

In [37]:
from sklearn.preprocessing import StandardScaler

scaler = StandardScaler()

X_train = scaler.fit_transform(X_train)
X_test = scaler.transform(X_test)

print(X_train.shape)
print(X_test.shape)

(40000, 392)
(10000, 392)


In [38]:
import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Dropout

In [39]:
model = Sequential([
    Dense(256, activation='relu',
          input_shape=(X_train.shape[1],)),
    Dropout(0.3),

    Dense(128, activation='relu'),
    Dropout(0.3),

    Dense(64, activation='relu'),

    Dense(1, activation='sigmoid')
])

/usr/local/lib/python3.12/dist-packages/keras/src/layers/core/dense.py:106: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


In [40]:
model.compile(
    optimizer='adam',
    loss='binary_crossentropy',
    metrics=[
        tf.keras.metrics.AUC(name='auc'),
        'accuracy'
    ]
)

In [41]:
history = model.fit(
    X_train,
    y_train,
    validation_split=0.2,
    epochs=10,
    batch_size=512
)

Epoch 1/10
63/63 ━━━━━━━━━━━━━━━━━━━━ 13s 87ms/step - accuracy: 0.9702 - auc: 0.6872 - loss: 0.1408 - val_accuracy: 0.9774 - val_auc: 0.8567 - val_loss: 0.0864
Epoch 2/10
63/63 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.9748 - auc: 0.8137 - loss: 0.0999 - val_accuracy: 0.9789 - val_auc: 0.8774 - val_loss: 0.0817
Epoch 3/10
63/63 ━━━━━━━━━━━━━━━━━━━━ 1s 8ms/step - accuracy: 0.9757 - auc: 0.8507 - loss: 0.0915 - val_accuracy: 0.9793 - val_auc: 0.8777 - val_loss: 0.0801
Epoch 4/10
63/63 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.9764 - auc: 0.8681 - loss: 0.0872 - val_accuracy: 0.9800 - val_auc: 0.8881 - val_loss: 0.0752
Epoch 5/10
63/63 ━━━━━━━━━━━━━━━━━━━━ 1s 8ms/step - accuracy: 0.9780 - auc: 0.8756 - loss: 0.0828 - val_accuracy: 0.9794 - val_auc: 0.8858 - val_loss: 0.0754
Epoch 6/10
63/63 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.9785 - auc: 0.8918 - loss: 0.0794 - val_accuracy: 0.9803 - val_auc: 0.8875 - val_loss: 0.0763
Epoch 7/10
63/63 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step 

In [42]:
!pip install mlflow -q

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 49.7/49.7 kB 3.4 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 50.5/50.5 kB 2.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 43.7/43.7 kB 2.1 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.6/12.6 MB 56.6 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.5/3.5 MB 73.6 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.7/1.7 MB 59.6 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.7/4.7 MB 51.6 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 147.8/147.8 kB 6.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 114.9/114.9 kB 8.8 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 212.0/212.0 kB 13.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 94.9/94.9 kB 9.9 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 132.2/132.2 kB 11.6 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 939.7/

In [43]:
import mlflow

print("MLflow berhasil diimport")

MLflow berhasil diimport


In [44]:
with mlflow.start_run():

    mlflow.log_param("epochs", 10)
    mlflow.log_param("batch_size", 512)

    mlflow.log_metric("accuracy", 0.9807)
    mlflow.log_metric("precision", 0.8421)
    mlflow.log_metric("recall", 0.3542)
    mlflow.log_metric("f1_score", 0.4987)
    mlflow.log_metric("roc_auc", 0.8898)

print("MLflow logging selesai")

2026/06/23 23:13:31 INFO mlflow.store.db.utils: Creating initial MLflow database tables...
2026/06/23 23:13:31 INFO mlflow.store.db.utils: Updating database tables


MLflow logging selesai


In [45]:
!ls

drive  mlflow.db  sample_data


In [46]:
!ls mlruns

ls: cannot access 'mlruns': No such file or directory


In [47]:
import mlflow

with mlflow.start_run():

    mlflow.log_param("epochs", 10)
    mlflow.log_param("batch_size", 512)

    mlflow.log_metric("accuracy", 0.9807)

    run_id = mlflow.active_run().info.run_id

print("Run ID:", run_id)

Run ID: a5f6df6c307447a3b99b1e4ee4b46901


## MLflow Tracking

MLflow digunakan untuk melakukan tracking parameter dan performa model. Parameter yang dicatat meliputi jumlah epoch dan batch size. Selain itu, metrik evaluasi seperti accuracy juga direkam untuk memudahkan proses monitoring dan reproduksi eksperimen.

Run ID yang dihasilkan dari proses tracking adalah a5f6df6c307447a3b99b1e4ee4b46901.